consul/consul.go

```diff
@@ -12,6 +13,7 @@ type Client struct {
 	c      *api.KV
 	folder string
 	codec  encoding.Codec
+	m      *sync.Mutex
```

Comment:
Did you find some Consul library documentation that says that `api.KV` is not thread-safe by itself? I don't remember seeing anything related to this back when I implemented this, but I could have overlooked it or maybe it has changed by now.

Reply:
No, I didn't. Also, I looked the `api.KV` code, I didn't find any unsafe writes. Race detector didn't detect anything too.  It's really strange bug. But race is there.